### DISCUSSION TITLE
Parallel computing of DP+PIMD

### DISCUSSION THREAD

**Discussion by zhangyu0104** [deepmodeling/deepmd-kit#2748] [Q&A]
Hi everyone,

I am trying to implement parallel computing of DP+PIMD on a single node using the SLURM queueing system. However, regardless of whether I allocate GPU or multiple CPU cores, it still can only run on a single CPU core. This is quite slow. I am wondering if there is a standard reference available to assist me in implementing parallel computing. Any help is greatly appreciated!

The version of DeePMD-kit is 2.1.1

Here is my submitting command:
```
i-pi input.xml > log.ipi &
sleep 5
dp_ipi water.json 1>log 2>err
wait
```
and the parameter file:
```
{
    "verbose":          true,
    "use_unix":         true,
    "port":             21344,
    "host":             "cnode004",
    "graph_file":       "frozen_model.pb",
    "coord_file":       "water_40.xyz",
    "atom_type" : {
        "O":           0,
        "H":          1
    }
}
```

**Comment by njzjz** [comment]
`dp_ipi` is a simple i-PI client that does not support MPI. You can consider using LAMMPS [`fix ipi`](https://docs.lammps.org/fix_ipi.html) or [`fix pimd/langevin`](https://docs.lammps.org/fix_pimd.html), which can be powered by LAMMPS MPI implementation.
